From stable-bounces@linux.kernel.org  Wed Nov 15 01:27:27 2006
From: maks@sternwelten.at
Date: Wed, 15 Nov 2006 00:59:02 GMT
Message-Id: <200611150059.kAF0x2xl009619@hera.kernel.org>
To: stable@kernel.org
Subject: cciss: fix iostat

From: Jens Axboe <jens.axboe@oracle.com>

cciss needs to call disk_stat_add() for iostat to work.

Signed-off-by: Jens Axboe <jens.axboe@oracle.com>
Signed-off-by: Linus Torvalds <torvalds@osdl.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 drivers/block/cciss.c |    6 ++++++
 1 file changed, 6 insertions(+)

--- linux-2.6.18.2.orig/drivers/block/cciss.c
+++ linux-2.6.18.2/drivers/block/cciss.c
@@ -1302,6 +1302,12 @@ static void cciss_softirq_done(struct re
 
 	complete_buffers(rq->bio, rq->errors);
 
+	if (blk_fs_request(rq)) {
+		const int rw = rq_data_dir(rq);
+
+		disk_stat_add(rq->rq_disk, sectors[rw], rq->nr_sectors);
+	}
+
 #ifdef CCISS_DEBUG
 	printk("Done with %p\n", rq);
 #endif				/* CCISS_DEBUG */
